sos/plugins/npm.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2014 Red Hat, Inc.,Poornima M. Kshirsagar <pkshiras@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+
+
+class Npm(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+    """
+    Get info about available npm modules
+    """
+
+    plugin_name = 'npm'
+    profiles = ('system',)
+
+    # in Fedora, Debian, Ubuntu the package is called npm
+    # in RHEL it's a software collection
+    # upstream doesn't have a npm package, it's just nodejs
+    # TODO: can we solve this? let's say: if there's npm command, run this plugin
+    packages = ('npm',)
```

Comment:
You solve the package name problem by defining an abstract base class (inherits from `Plugin`), and then creating concrete subclasses that alter the required attributes, e.g. something like:

```python
class RedHatNpm(Npm, RedHatPlugin):
        packages = ('funky-npm',)

class UbuntuNpm(Npm, UbuntuPlugin):
        packages = ('cool-npm',)

```

All the common code can normally stay in the base class and the subclasses effectively just set metadata for it to interpret.